release/release-debian

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash -eu
+#
+# release-debian
+#
+# A release script that publishes packages to a debian repository.
+#
+# Arguments are described here. Most arguments have an equivalent envvar.
+#
+# -d src     RELEASE_DEBIAN_SOURCE=src  Debian source repository
+# -r repo    RELEASE_DEBIAN_REPO=repo   Generated Debian repository
+#
+# -q         RELEASE_QUIET=1            Make output more quiet
+# -x         RELEASE_TRANSACTION=1      SIGSTOP before pushing the dist-git commit
+# -v         RELEASE_VERBOSE=1          Make output more verbose
+# -z         RELEASE_CHECK=1            Check credentials only
+#
+
+set -eu
+
+TRANSACTION=${RELEASE_TRANSACTION:-0}
+CHECK=${RELEASE_CHECK:-0}
+QUIET=${RELEASE_QUIET:-0}
+VERBOSE=${RELEASE_VERBOSE:-0}
+
+TAG=${RELEASE_TAG:-}
+DEBIAN_SOURCE=${RELEASE_DEBIAN_SOURCE:-}
+DEBIAN_REPO=${RELEASE_DEBIAN_REPO:-debian-apt}
+
+usage()
+{
+    echo "usage: release-debian [-qvxz] [-t tag] [-d source] [-r repository]" >&2
+    exit ${1-2}
+}
+
+trace()
+{
+    if [ $QUIET -eq 0 ]; then
+        echo "> $@" >&2
+    fi
+}
+
+message()
+{
+    echo "release-debian: $@" >&2
+}
+
+check()
+{
+    true
+}
+
+prepare()
```

Comment:
Is this function idempotent? If it's called again, on exactly the same input, will it skip it's task?